graphql/resolvers.go

```diff
@@ -435,7 +435,8 @@ func (r *taskResolver) CanOverrideDependencies(ctx context.Context, at *restMode
 		RequiredLevel: evergreen.TasksAdmin.Value,
 		Resource:      *at.ProjectId,
 	}
-	if len(at.DependsOn) > 0 && currentUser.HasPermission(requiredPermission) {
+	if len(at.DependsOn) > 0 && (currentUser.HasPermission(requiredPermission) ||
```

Comment:
Ors are evaluated in order. There's no reason to look up if the user has permission (which is a DB query) if it's a patch requester, so we should put this as the latter part of the clause.